maplibre/src/coords.rs

```diff
@@ -31,6 +31,10 @@ const fn create_zoom_bounds<const DIM: usize>() -> [u32; DIM] {
     result
 }
 
+/// Represents the position of a node within a quad tree. It stores the number of one of the four
+/// squares for each subdivision of the quad tree from zoom level 0.
+///
+/// TODO: Would it be useful to optimize the quadkey and store the keys on 3 bits instead of 8?
```

Comment:
Yes that definitely would be an optimization. https://github.com/maplibre/maplibre-rs/issues/90

But wouldn't it be 2 bit?

Reply:
I was wrong with the 3 bits idea. I didn't understand that the first 8 bits of the Quadkey store the zoom level. 

Shouldn't the key be `let mut key = [0u8; MAX_ZOOM+1];` if we store the zoom level in the first byte?

Anyway if we store the zoom level separatly we indeed only need 2 bits for NW, NE, SW and SE.

Should I update #90 with `2 bits * MAX_ZOOM + 8 bits for zoom level`?

Reply:
You are right :)

Reply:
Yes you can change it

Reply:
Just FYI, I was hacking on an optimized tile ID storage a while back - see https://github.com/nyurik/planetiler-rs/blob/main/experiments/src/tile_id.rs -- essentially its a single u32 that can store all z/x/y for z in `0..=15`.  I might create a crate at some point that has all the basic ID features.

Reply:
I think for this crate we want +15 zoom, but maybe it would be cool to have an enum between both representations.